flake.nix

```diff
@@ -0,0 +1,100 @@
+{
+  description = "LSP-AI: Open-source language server for AI-powered functionality";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+    flake-utils.url = "github:numtide/flake-utils";
+    rust-overlay.url = "github:oxalica/rust-overlay";
+  };
+
+  outputs =
+    {
+      self,
+      nixpkgs,
+      flake-utils,
+      rust-overlay,
+    }:
+    flake-utils.lib.eachDefaultSystem (
+      system:
+      let
+        overlays = [ (import rust-overlay) ];
+        pkgs = import nixpkgs {
+          inherit system overlays;
+          config.allowUnfree = true;
+        };
+        rustVersion = pkgs.rust-bin.stable.latest.default;
+
+        commonBuildInputs = with pkgs; [
+          openssl
+          zlib
+        ];
+
+        commonNativeBuildInputs = with pkgs; [
+          pkg-config
+          cmake
+          rustVersion
+          perl
+        ];
+
+      in
+      rec {
+        packages = rec {
+          default = pkgs.rustPlatform.buildRustPackage {
+            pname = "lsp-ai";
+            version = "0.7.0";
+
+            src = ./.;
+
+            cargoLock = {
+              lockFile = ./Cargo.lock;
+              allowBuiltinFetchGit = true;
+            };
+
+            nativeBuildInputs = commonNativeBuildInputs;
+            buildInputs = commonBuildInputs;
+
+            doCheck = false;
```

Comment:
Currently disables all testing. See [nixpkgs](https://github.com/ProjectInitiative/nixpkgs/blob/c7a5abc9129f4de313a645c2e9a81f760dad72f0/pkgs/by-name/ls/lsp-ai/package.nix#L23) implementation to see how to disable individual tests. As this nix flake is most likely not going to be integrated into any CI, I think it is okay to disable them for now since the goal is just for folks who want to pull and have a dev env/build from source avenue. Suggestions welcome.